crates/primitives/src/storage.rs

```diff
@@ -1,12 +1,28 @@
 use super::{H256, U256};
-use reth_codecs::{main_codec, Compact};
+use reth_codecs::Compact;
 
 /// Account storage entry.
 #[derive(Debug, Default, Clone, PartialEq, Eq)]
-#[main_codec]
 pub struct StorageEntry {
     /// Storage key.
     pub key: H256,
     /// Value on storage key.
     pub value: U256,
 }
+
+impl Compact for StorageEntry {
```

Comment:
Let's add a note on the reason why we manually implemented it (ie what you describe in the PR